bundles/org.openhab.binding.solarforecast/src/main/java/org/openhab/binding/solarforecast/internal/forecastsolar/ForecastSolarObject.java

```diff
@@ -0,0 +1,346 @@
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class ForecastSolarObject implements SolarForecast {
+    private final Logger logger = LoggerFactory.getLogger(ForecastSolarObject.class);
+    private final TreeMap<ZonedDateTime, Double> wattHourMap = new TreeMap<>();
+    private final TreeMap<ZonedDateTime, Double> wattMap = new TreeMap<>();
+    private final DateTimeFormatter dateInputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+
+    private DateTimeFormatter dateOutputFormatter = DateTimeFormatter
+            .ofPattern(SolarForecastBindingConstants.PATTERN_FORMAT).withZone(ZoneId.systemDefault());
+    private ZoneId zone = ZoneId.systemDefault();
+    private Optional<String> rawData = Optional.empty();
+    private Instant expirationDateTime;
+    private String identifier;
+
+    public ForecastSolarObject(String id) {
+        expirationDateTime = Instant.now().minusSeconds(1);
+        identifier = id;
+    }
+
+    public ForecastSolarObject(String id, String content, Instant expirationDate) throws SolarForecastException {
+        expirationDateTime = expirationDate;
+        identifier = id;
+        if (!content.isEmpty()) {
+            rawData = Optional.of(content);
+            try {
+                JSONObject contentJson = new JSONObject(content);
+                JSONObject resultJson = contentJson.getJSONObject("result");
+                JSONObject wattHourJson = resultJson.getJSONObject("watt_hours");
+                JSONObject wattJson = resultJson.getJSONObject("watts");
+                String zoneStr = contentJson.getJSONObject("message").getJSONObject("info").getString("timezone");
+                zone = ZoneId.of(zoneStr);
+                dateOutputFormatter = DateTimeFormatter.ofPattern(SolarForecastBindingConstants.PATTERN_FORMAT)
+                        .withZone(zone);
+                Iterator<String> iter = wattHourJson.keys();
+                // put all values of the current day into sorted tree map
+                while (iter.hasNext()) {
+                    String dateStr = iter.next();
+                    // convert date time into machine readable format
+                    try {
+                        ZonedDateTime zdt = LocalDateTime.parse(dateStr, dateInputFormatter).atZone(zone);
+                        wattHourMap.put(zdt, wattHourJson.getDouble(dateStr));
+                        wattMap.put(zdt, wattJson.getDouble(dateStr));
+                    } catch (DateTimeParseException dtpe) {
+                        logger.warn("Error parsing time {} Reason: {}", dateStr, dtpe.getMessage());
+                        throw new SolarForecastException(this,
+                                "Error parsing time " + dateStr + " Reason: " + dtpe.getMessage());
+                    }
+                }
+            } catch (JSONException je) {
+                logger.debug("Error parsing JSON response {} - {}", content, je.getMessage());
```

Comment:
This could be logged by the handler when setting Thing status. This would centralize the error handling.

Reply:
Offline Status Details adapted

Reply:
The redundant local logging is still here. Since you are now using the exception message for setting the Thing status, it will already be logged.

Reply:
removed